### DISCUSSION TITLE
How to speed up waiting time until Tune-In radio actually plays?

### DISCUSSION THREAD

**Discussion by veitw** [music-assistant/support#897] [Q&A]
Using the Tune-in provider, after starting to play a radio station, it takes 25 to 35 seconds until the music actually starts playing.

Using Media Browser, Radio Browser radio stations start almost instantly when playing to the original `media_player` entities, but show the same delay as in MA's Tune-in when playing to `media_player.mass_*` entities.

Is it possible to shorten this delay?
Maybe some kind of buffering, where one might configure a buffer size?

Players: ESP32 SoCs running ESPhome, registered as `media_player`s to HA using ESPhome's `api` and `media_player` components.

**Comment by pietro4711** [comment]
I have the exact same setup - the "time to play" is so much longer with the mass_*-Players.
Also if a play announcements while music is playing the time till resuming is very long.

**Comment by preceus** [comment]
The same problem with ESP, I tried everything I could - when using MASS, the delay is 25 seconds. There is no delay when playing mp3 files. But even if you insert the url in the form of m3u, there is a delay of 25 seconds. When switching to the wav format in the queue menu settings, radio playback is instantaneous, but hissing every second.

**Comment by marcelveldt** [comment]
I really wouldn't recommend using ESPHome as an audio player other than some announcements. My recommendation would be to run squeezelite on it together with the slimproto integration in HA. That will give a much better experience and audio quality.

**Reply by wuast94** [reply]
i have the same issue, using youtube music and sonos, booth dlna and sonos native. it takes 10-30 seconds to respond

**Reply by OzGav** [reply]
The previous response was in relation to the use of ESP devices. I am assuming from your post you are not using those. You will need to check the logs as we aren’t having other reports of this issue.

**Comment by OzGav** [comment]
Outdated info as it relates to v1
